src/frontend/komponenter/Fagsak/Vilkår/BehandleVilkårProvider.tsx

```diff
@@ -81,6 +91,14 @@ const behandlingVilkårReducer = (state: IState, action: IAction): IState => {
                         ? VedtakResultat.AVSLÅTT
                         : VedtakResultat.INNVILGET,
             };
+        case actions.SETT_BEGRUNNELSE:
+            return {
+                ...state,
+                begrunnelse: state.begrunnelse.valideringsFunksjon({
```

Comment:
Shouldnt the validation function return a new IFelt? Here we only update the value, and not the errormessage and so forth.

Reply:
I am a bit confused. Shouldn't we do the validation here? You have done the same in the OpprettBehandlingProvider.tsx...

Reply:
No, if you just expand your validation function to return IFelt you basically do validation when you set the field. And in our submit function we can iterate through all the fields and if one is not validated or has faults, we do not submit and set the visFeilmeldinger flagg.